app/src/main/java/com/example/habitsmasher/ui/dashboard/HabitListFragment.java

```diff
@@ -171,4 +217,35 @@ public void onFailure(@NonNull Exception e) {
     @Override
     public void onDestroyView() { super.onDestroyView();
     }
+
+
+    public void updateAfterEdit(String title, String reason, Date date, int pos) {
+        //String oldHabitTitle = _habitList.getHabitList().get(pos).getTitle();
+        Long habitId = _habitItemAdapter._snapshots.get(pos).getHabitId();
+        //_habitList.editHabit(title, reason, date, pos)
+        // storing in database
+        HashMap<String, Object> habitData = new HashMap<>();
+        habitData.put("title", title);
+        habitData.put("reason", reason);
+        habitData.put("date", date);
+        habitData.put("habitId", habitId);
+        //collectionReference.document(habitId.toString()).update(habitData);
+        collectionReference.document(habitId.toString())
+                .set(habitData)
+                .addOnSuccessListener(new OnSuccessListener<Void>() {
+                    @Override
+                    public void onSuccess(Void unused) {
+                        Log.d(TAG, "Data successfully added.");
+                    }
+                })
+                .addOnFailureListener(new OnFailureListener() {
+                    @Override
+                    public void onFailure(@NonNull Exception e) {
+                        Log.d(TAG, "Data failed to be added." + e.toString());
+                    }
+                });
+        _editButton.setVisibility(View.INVISIBLE);
+        _deleteButton.setVisibility(View.INVISIBLE);
+        _habitItemAdapter.notifyDataSetChanged();
```

Comment:
If possible, instead of calling `notifyDataSetChanged()`, we should be specific about what's happening. You can do action-specific notifies such as add, delete, or edit.